mocks/mockServer.go

```diff
@@ -120,6 +122,19 @@ func (m *MockServer) WellKnownHandler(w http.ResponseWriter, _ *http.Request) {
 	_, _ = w.Write(payload)
 }
 
+// ClientCredentialsTokenHandler is the http handler which serves the /oauth2/token endpoint. It returns a token without claims.
+func (m *MockServer) ClientCredentialsTokenHandler(w http.ResponseWriter, r *http.Request) {
+	buf := new(bytes.Buffer)
+	_, _ = buf.ReadFrom(r.Body)
+	newStr := buf.String()
+	if strings.Contains(newStr, "grant_type=client_credentials") && strings.Contains(newStr, "client_id="+m.Config.ClientID) {
+		payload, _ := json.Marshal(tokenResponse{
+			Token: "eyJhbGciOiJIUzI1NiJ9.e30.ZRrHA1JJJW8opsbCGfG_HACGpVUMN_a9IV7pAx_Zmeo",
+		})
+		_, _ = w.Write(payload)
+	}
```

Comment:
```suggestion
	grantType := r.PostFormValue("grant_type")
	clientID := r.PostFormValue("client_id")
	if grantType != "client_credentials" || clientID != m.Config.ClientID {
		w.WriteHeader(http.StatusUnauthorized)
		return
	}

	_ = json.NewEncoder(w).Encode(tokenResponse{
		Token: "eyJhbGciOiJIUzI1NiJ9.e30.ZRrHA1JJJW8opsbCGfG_HACGpVUMN_a9IV7pAx_Zmeo",
	})
```

Reply:
The rest looks good 👍🏼

Reply:
Hi Felix, in case you want to return `401` i have to rename the handler from clientCredentialsTokenHandler to tokenHandler

Reply:
Why? It's still only client_credentials.
The previous implementation would've returned status 200 with no token. I just changed the status code to 401 if no token is returned.

Reply:
you're right. In future in order to support other grant types they have to be combined in one handler, as the route is the same. My previous assumption was, that I'm able to attach multiple handlers to the ServeMux -> but you can multiplex to different subhandlers by request route, which isn't the case here.